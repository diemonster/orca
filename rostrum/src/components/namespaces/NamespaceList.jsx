import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { namespaceList } from '../../actions/namespaces';

import NamespaceListItem from './NamespaceListItem';

class NamespaceList extends React.Component {
  componentDidMount() {
    const { dispatchNamespaceList } = this.props;

    dispatchNamespaceList();
  }

  render() {
    const { namespaceObjects } = this.props;

    if (!namespaceObjects) {
      return (
        <div className="namespace-list-container">
          <h2>List Namespaces</h2>
          <p>Loading namespaces...</p>
        </div>
      );
    }

    const namespaceListItems = namespaceObjects.map(namespace => (
      <NamespaceListItem key={namespace.name} namespace={namespace.name} phase={namespace.status} />
    ));

    return (
      <div className="namespace-list-container">
        <h2>List Namespaces</h2>
        <ul className="namespace-list">
          {namespaceListItems}
        </ul>
      </div>
    );
  }
}

NamespaceList.propTypes = {
  dispatchNamespaceList: PropTypes.func.isRequired,
  namespaceObjects: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      status: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

const mapStateToProps = state => ({
  namespaceObjects: state.namespace.namespaceObjects,
});

const mapDispatchToProps = dispatch => ({
  dispatchNamespaceList: () => dispatch(namespaceList()),
});

export default connect(mapStateToProps, mapDispatchToProps)(NamespaceList);
