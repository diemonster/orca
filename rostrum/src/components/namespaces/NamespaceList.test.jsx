import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { NamespaceList } from './NamespaceList';

Enzyme.configure({ adapter: new Adapter() });

describe('components', () => {
  describe('NamespaceList', () => {
    it('should render itself and its subcomponents when namespaceObjects === 0', () => {
      const props = {
        dispatchNamespaceList: jest.fn(),
        namespaceObjects: [],
      };

      const namespaceList = shallow(<NamespaceList {...props} />);

      expect(namespaceList).toMatchSnapshot();
      expect(namespaceList.find('div').hasClass('namespace-list-container')).toBe(true);
      expect(namespaceList.find('h2').text()).toEqual('List Namespaces');
      expect(namespaceList.find('p').text()).toEqual('Loading namespaces...');
    });

    it('should render itself and subcomponents when namespaceObjects.length > 0', () => {
      const props = {
        dispatchNamespaceList: jest.fn(),
        namespaceObjects: [
          { name: 'namespace1', status: 'Active' },
          { name: 'namespace2', status: 'Terminating' },
        ],
      };

      const namespaceList = shallow(<NamespaceList {...props} />);

      expect(namespaceList.find('div').hasClass('namespace-list-container')).toBe(true);
      expect(namespaceList.find('h2').text()).toEqual('List Namespaces');
      expect(namespaceList.find('ul').hasClass('namespace-list')).toBe(true);

      const listItems = namespaceList.find('ul').children();
      expect(listItems.length).toEqual(2);
      expect(listItems.at(0).props().namespace).toEqual('namespace1');
      expect(listItems.at(0).props().phase).toEqual('Active');
      expect(listItems.at(1).props().namespace).toEqual('namespace2');
      expect(listItems.at(1).props().phase).toEqual('Terminating');
    });
  });
});
