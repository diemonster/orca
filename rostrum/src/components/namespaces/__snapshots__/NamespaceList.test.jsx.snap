// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components NamespaceList should render itself and its subcomponents when namespaceObjects === 0 1`] = `
<div
  className="namespace-list-container"
>
  <h2>
    List Namespaces
  </h2>
  <p>
    Loading namespaces...
  </p>
</div>
`;
